happi/audit.py

```diff
@@ -59,6 +61,60 @@ def check_name_match_id(result: SearchResult) -> None:
     assert id_field == name_field, f'id: {id_field} != name: {name_field}'
 
 
+def check_wait_connection(result: SearchResult) -> None:
+    """
+    Check if all the signals on the device can connect to their PV's
+
+    This can take a long time, consider filtering your query before running
+    """
+    dev = result.get()
+
+    assert (hasattr(dev, 'wait_for_connection') and
+            callable(getattr(dev, 'wait_for_connection'))), \
+           ('device has no wait_for_connection method, and is likely '
+            'not an ophyd device')
+
+    try:
+        dev.wait_for_connection(timeout=5)
+    except TimeoutError as te:
+        # If we encounter a timeout error, gather some more details stats
+        if hasattr(dev, "walk_signals"):
+            sigs = list(sig.item for sig in dev.walk_signals())
+            conn_sigs = sum(sig.connected for sig in sigs)
+
+        raise ValueError(f'{conn_sigs}/{len(sigs)} signals connected. \n'
```

Comment:
`conn_sigs` and `sigs` can be unbound here if `walk_signals` doesn't exist on `dev`

Should `walk_signals` existence be assumed, instead, as getting here would indicate an ophyd v1 device?

Reply:
Ah, good point.  I think it makes sense to assume ohpyd v1 at this point, if a timeout error is thrown then surely we've reached the `Device.wait_for_connection` method..